### PR TITLE
feat: support DCS and batch queries

### PR DESCRIPTION
**Pull request by jgrumboe**
This PR adds two things:
- DCS support for resource_name mappings
- Batch metric queries

I know you can't test the implementation, locally it works like a charm and I also use this version in our setup already.
But I would appreciate your feedback on the code itself, as I'm no python programmer and most of my work is googling, copy/pasting and modifying. 🙂

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -173,57 +193,95 @@ def generate_prometheus_metrics(metrics):
...
+                    "period": "300",
+                    "filter": "average"
+                }
+
+                # Reset metrics_set for next batch
+                metrics_set = []
+
+
+                # Wait for 100 milliseconds before every request (needed to don't overload OTC API)
+                time.sleep(0.1)
```
Comment (tiagoReichert) [inline app/main.py:250]: I know the previous version didn't have it, but what do you think about adding a exponential backoff retry strategy?

File: app/main.py [hunk 2]
```diff
@@ -173,57 +193,95 @@ def generate_prometheus_metrics(metrics):
...
+                                                                            resource_id=dimensions_value,
+                                                                            resource_name=resource_name).set(mr['datapoints'][0]['average'])
+    
+                                logging.debug("{0} for '{1}={2}' ({3}) at {4} : {5}".format(full_metric_name, dimensions_name,
+                                                                                dimensions_value, resource_name,
+                                                                                datetime.fromtimestamp(mr['datapoints'][0]['timestamp'] / 1000.0),
+                                                                                mr['datapoints'][0]['average']))
+
+
+                elif r.status_code == 401:
```
Comment (tiagoReichert) [inline app/main.py:277]: Would have to find out wich return code the API has for throttling and the exponential backoff retry strategy could be added here.

**Review by tiagoReichert — COMMENTED** [review]
Can you please check if it makes sense to add a exponential backoff retry strategy?

File: app/main.py (hunk 1 above)
Comment (jgrumboe) [reply app/main.py:250]: Sounds like a good idea. Will try to investigate.

File: app/main.py (hunk 2 above)
Comment (jgrumboe) [reply app/main.py:277]: According to the normal API docs https://docs.otc.t-systems.com/en-us/api/apiug/apig-en-api-180328012.html and CloudEye docs  https://docs.otc.t-systems.com/api/ces/ces_03_0055.html it seems that there's no response code specific to throttling.

Comment (jgrumboe) [reply app/main.py:277]: https://docs.otc.t-systems.com/api/ces/ces_03_0023.html mentions an HTTP 429 for too many requests.

File: app/main.py (hunk 1 above)
Comment (jgrumboe) [reply app/main.py:250]: > Can you please check if it makes sense to add a exponential backoff retry strategy?

@tiagoReichert 
That sounds promising: https://stackoverflow.com/a/35504626
What do you think?

**Comment by saibotma** [comment]
Got this PR merged somewhere? Maybe in a forked repository?

**Comment by jgrumboe** [comment]
@saibotma yes, you can just use the source branch of this PR

File: app/main.py (hunk 1 above)
Comment (tiagoReichert) [reply app/main.py:250]: @jgrumboe Yes, I think it makes sense. Can you add this to the PR?

**Comment by jgrumboe** [comment]
I won't work on this PR as I'm not using OTC anymore.
Should I close it?

**Pull request closed without merging**